src/indicators/drought/ecmwf_seasonal.py

```diff
@@ -0,0 +1,301 @@
+        self.year_max = year_max
+        self.cds_name = cds_name
+        self.dataset = dataset
+        self.dataset_variable_name = dataset_variable_name
+
+    def _download(
+        self,
+        country_name: str,
+        country_iso3: str,
+        area: Area,
+        version: int,
+        year: int,
+        month: int = None,
+        leadtime: int = None,
+        use_cache: bool = True,
+    ):
+        filepath = self._get_raw_filepath(
+            country_name=country_name,
+            country_iso3=country_iso3,
+            version=version,
+            year=year,
+            month=month,
+        )
+        # If caching is on and file already exists, don't download again
+        if use_cache and filepath.exists():
+            logger.debug(
+                f"{filepath} already exists and cache is set to True, skipping"
+            )
+            return filepath
+        Path(filepath.parent).mkdir(parents=True, exist_ok=True)
+        logger.debug(f"Querying for {filepath}...")
+        logger.debug(f"{self._get_query(area=area,version=version,year=year,month=month,leadtime=leadtime, )}")
+        CDSAPI_CLIENT.retrieve(
+            name=self.cds_name,
+            request=self._get_query(
+                area=area,
+                version=version,
+                year=year,
+                month=month,
+                leadtime=leadtime,
+            ),
+            target=filepath,
+        )
+        logger.debug(f"...successfully downloaded {filepath}")
+        # Wait 2 seconds between requests or else API hangs
+        # TODO make sure this actually works
+        time.sleep(2)
+        return filepath
+
+    def _get_raw_filepath(
+        self,
+        country_name: str,
+        country_iso3: str,
+        version: int,
+        year: int,
+        month: int = None,
+        leadtime: int = None,
+    ):
+        directory = (
+            RAW_DATA_DIR
+            / country_name
+            / ECMWF_SEASONAL_DIR
+            / self.cds_name
+        )
+        filename = f"{country_iso3}_{self.cds_name}_v{version}_{year}"
+        if month is not None:
+            filename += f"-{str(month).zfill(2)}"
+        if leadtime is not None:
+            filename += f"_lt{str(leadtime).zfill(2)}d"
+        filename += ".grib"
+        return directory / Path(filename)
+
+    def _get_query(
+        self,
+        area: Area,
+        version: int,
+        year: int,
+        month: int = None,
+        leadtime: int = None,
+    ) -> dict:
+        query = {
+            "variable": "total_precipitation",
+            'originating_centre': 'ecmwf',
+            'system': version,
+            "format": "grib",
+            self.dataset_variable_name: self.dataset,
+            "year": str(year),
+            "month": [str(x).zfill(2) for x in range(1, 13)]
+            if month is None
+            else str(month).zfill(2),
+            "leadtime_month":[str(x) for x in range(1,7)]
```

Comment:
why 7?

Reply:
Good point. This are the leadtimes, now set it as a global var